internal/iso_packaging/iso_packaging.go

```diff
@@ -0,0 +1,241 @@
+package iso_packaging
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/project-flotta/osbuild-operator/api/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+const (
+	templateCommand string = "isopackage %s -ks %s --upload-target"
+	kickstart       string = "init.ks"
+)
+
+var (
+	jobTTLafterFinish int32 = 100   // Job will be deleted after finished with this
+	deadlineSeconds   int64 = 36000 // Job will be terminated after a hour.
+)
+
+// Builder is a struct that manages a build to package an iso
+type Builder struct {
+	client      client.Client
+	jobSpec     *batchv1.Job
+	build       *v1alpha1.OSBuild
+	buildConfig *v1alpha1.OSBuildEnvConfig
+	baseImage   string
+}
+
+func NewBuilderJob(client client.Client, build *v1alpha1.OSBuild, buildConfig *v1alpha1.OSBuildEnvConfig, image string) (*Builder, error) {
+
+	if build == nil || buildConfig == nil {
+		return nil, fmt.Errorf("Invalid config added")
+	}
+
+	return &Builder{
+		client:      client,
+		jobSpec:     nil,
+		build:       build,
+		buildConfig: buildConfig,
+		baseImage:   image,
+	}, nil
+}
+
+func (b *Builder) Start() error {
+	sourceISO := b.build.Status.ComposerIso
+	if sourceISO == "" {
+		return fmt.Errorf("Cannot parse invalid iso image")
+	}
+	_, err := url.Parse(sourceISO)
+	if err != nil {
+		return err
+	}
+
+	if b.build.Spec.Kickstart == nil {
+		return fmt.Errorf("Kickstart is not defined, skipping")
+	}
+
+	command := fmt.Sprintf(templateCommand, b.build.Status.ComposerIso, kickstart)
+
+	jobSpec := &batchv1.Job{
```

Comment:
Consider moving this struct to a separate method to improve this one's readability.